### PR TITLE
Remove contiguous from passes for reshapes

### PR DESCRIPTION
**Pull request by TedThemistokleous**
- Remove contiguous from simplify algebra passes for reshape op
- Remove contiguous from fuse_pointwise for reshape op
- related to #2099 & #2190 involving reshape_lazy

### CODE REVIEW

**Comment by TedThemistokleous** [comment]
@kahmed10  not sure if this undoes your branch with adding contiguous. What was that PR again?

**Review by CharlieL7 — COMMENTED** [review]
What's the status on this PR? Waiting on Khalique to respond or additional changes?

**Comment by TedThemistokleous** [comment]
> What's the status on this PR? Waiting on Khalique to respond or additional changes?

Not sure, I haven't seen much movement on it. I dont intend to add anything else to this

**Comment by TedThemistokleous** [comment]
Rebased off develop and added changes based on @umangyadav pertaining to reshape since this is for reshape_lazy changes done previously

@CharlieL7  if you want to take a look/have comments this should be fine to review now

One question I do have for @umangyadav   pruning other reshapes I suppose we'll try to prune based on this list and then handle each matcher case by case? I'll create another PR for these to segment the work up if need be.
```
const auto& reshaper_names()
{
    // clang-format off
    static const std::unordered_set<std::string> names = {
        "flatten",
        "reshape",
        "contiguous",
        "squeeze",
        "unsqueeze"
    };
    // clang-format on
    return names;
}
```

**Review by CharlieL7 — APPROVED** [review]
LGTM, CI fixes needed

**Review by pfultz2 — APPROVED** [review]

File: test/simplify_algebra_test.cpp [hunk 1]
```diff
@@ -1897,12 +1897,17 @@ TEST_CASE(simplify_split_add_relu_reshape)
         auto concatb = m2.add_instruction(b, concat);
         auto sum     = m2.add_instruction(migraphx::make_op("add"), input, concatb);
         auto relu    = m2.add_instruction(migraphx::make_op("relu"), sum);
-        auto rsp     = m2.add_instruction(migraphx::make_op("reshape", {{"dims", {3, 8}}}), relu);
```
Comment (umangyadav) [inline test/simplify_algebra_test.cpp:1900]: Wondering what change caused this reshape to split into two.

**Comment by causten** [comment]
@TedThemistokleous  Format has a problem after that I can merge

**Pull request merged**